p2panda-rs/src/secret_group/error.rs

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use thiserror::Error;
+
+/// Custom error types for `SecretGroup`.
+#[derive(Error, Debug)]
+#[allow(missing_copy_implementations)]
+pub enum SecretGroupError {
+    /// MLS commit message was expected to contain a welcome message as well.
+    #[error("commit does not contain welcome message")]
+    WelcomeMissing,
```

Comment:
This would an OpenMLS library error. Creating a Commit message that covers Add proposals should always yield a Welcome message.

Reply:
This error is used by the `new_from_welcome` method here to throw if someone wrongly passed in another commit (update or remove etc.) since at this stage a "commit" is relatively generic in how our API is designed: https://github.com/p2panda/p2panda/blob/mls-core/p2panda-rs/src/secret_group/group.rs#L78